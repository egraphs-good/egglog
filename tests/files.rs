use egg_smol::*;

fn test_program(program: &str, message: &str) {
    let mut egraph = EGraph::default();
    match egraph.parse_and_run_program(program) {
        Ok(msgs) => {
            for msg in msgs {
                log::info!("  {}", msg);
            }
        }
        Err(err) => panic!("{}: {err}", message),
    }
}

fn run(path: &str) {
    let _ = env_logger::builder().is_test(true).try_init();
    let program = std::fs::read_to_string(path).unwrap();
    test_program(&program, "Top level error");

    let egraph = EGraph::default();
    let program_str = egraph
        .parse_program(&program)
        .unwrap()
        .into_iter()
        .map(|x| x.to_string())
        .collect::<Vec<String>>()
        .join("\n");
    test_program(
        &program_str,
        &format!(
            "Program:\n{}\n ERROR after parse, to_string, and parse again.",
            program_str
        ),
    );
}

// include the tests generated from the build script
include!(concat!(std::env!("OUT_DIR"), "/files.rs"));
