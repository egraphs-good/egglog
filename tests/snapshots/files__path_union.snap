---
source: tests/files.rs
expression: serialized
---
nodes:
  Unit-0:
    op: ()
    children: []
    eclass: Unit-0
    cost: 0
  path-7659469028595837896:
    op: path
    children:
      - mk-11743562013128004906
      - mk-10912160959110460649
    eclass: Unit-0
    cost: 1
  path-5871781006564002453:
    op: path
    children:
      - mk-5871781006564002453
      - mk-11743562013128004906
    eclass: Unit-0
    cost: 1
  path-15318938057191675792:
    op: path
    children:
      - mk-17615343019692007359
      - mk-16783941965674463102
    eclass: Unit-0
    cost: 1
  path-5996666920560749382:
    op: path
    children:
      - mk-11743562013128004906
      - mk-16783941965674463102
    eclass: Unit-0
    cost: 1
  path-11743562013128004906:
    op: path
    children:
      - mk-5871781006564002453
      - mk-10912160959110460649
    eclass: Unit-0
    cost: 1
  path-5040379952546458196:
    op: path
    children:
      - mk-5871781006564002453
      - mk-16783941965674463102
    eclass: Unit-0
    cost: 1
  edge-5871781006564002453:
    op: edge
    children:
      - mk-5871781006564002453
      - mk-11743562013128004906
    eclass: Unit-0
    cost: 1
  edge-7659469028595837896:
    op: edge
    children:
      - mk-11743562013128004906
      - mk-17615343019692007359
    eclass: Unit-0
    cost: 1
  edge-15318938057191675792:
    op: edge
    children:
      - mk-10912160959110460649
      - mk-16783941965674463102
    eclass: Unit-0
    cost: 1
  i64-5871781006564002453:
    op: "1"
    children: []
    eclass: i64-5871781006564002453
    cost: 0
  mk-5871781006564002453:
    op: mk
    children:
      - i64-5871781006564002453
    eclass: "0"
    cost: 1
  i64-11743562013128004906:
    op: "2"
    children: []
    eclass: i64-11743562013128004906
    cost: 0
  mk-11743562013128004906:
    op: mk
    children:
      - i64-11743562013128004906
    eclass: "1"
    cost: 1
  i64-17615343019692007359:
    op: "3"
    children: []
    eclass: i64-17615343019692007359
    cost: 0
  mk-17615343019692007359:
    op: mk
    children:
      - i64-17615343019692007359
    eclass: "2"
    cost: 1
  i64-16783941965674463102:
    op: "6"
    children: []
    eclass: i64-16783941965674463102
    cost: 0
  mk-16783941965674463102:
    op: mk
    children:
      - i64-16783941965674463102
    eclass: "4"
    cost: 1
  i64-10912160959110460649:
    op: "5"
    children: []
    eclass: i64-10912160959110460649
    cost: 0
  mk-10912160959110460649:
    op: mk
    children:
      - i64-10912160959110460649
    eclass: "2"
    cost: 1
root_eclasses: []
class_data:
  Unit-0:
    type: Unit
  "1":
    type: Node
  "2":
    type: Node
  "0":
    type: Node
  "4":
    type: Node
  i64-5871781006564002453:
    type: i64
  i64-11743562013128004906:
    type: i64
  i64-17615343019692007359:
    type: i64
  i64-16783941965674463102:
    type: i64
  i64-10912160959110460649:
    type: i64

